src/time/format/formatter.cr

```diff
@@ -147,10 +147,14 @@ struct Time::Format
       nanoseconds
     end
 
-    def second_fraction?(fraction_digits = nil)
-      unless time.nanosecond == 0 || fraction_digits == 0
-        char '.'
-        second_fraction
+    def second_fraction?(fraction_digits : Int = 0)
```

Comment:
This method nameshould be deprecated. The intention of the `?` was to express that the behaviour depends on the value. Now it doesn't, it's fully specified by the arguments. So it should just be `#second_fraction` (and `fraction_digits = 9` to keep the default behaviour)

Reply:
I just changed the default to 9 to be more compatible with the previous behaviour. But I think the name is fine, because it doesn't even print the `.` when digits = 0. Also, there is another method `second_fraction` and this name also matches with the method in the parser that expects optional decimals.